File: app/src/main/java/com/techventus/wikipedianews/WikiApplication.java
package com.techventus.wikipedianews;

import android.app.Application;


//import androidx.multidex.BuildConfig;

import java.util.Calendar;
import java.util.Date;
import com.techventus.wikipedianews.logging.Logger;
import com.techventus.wikipedianews.logging.Toaster;
import com.techventus.wikipedianews.manager.PreferencesManager;

/**
 * Created by josephmalone on 16-06-23.
 */
public class WikiApplication extends Application
{
	private static final String TAG = WikiApplication.class.getSimpleName();
	private static WikiApplication mWikiApp;

	public static WikiApplication getInstance()
	{
		return mWikiApp;
	}

	@Override
	public void onCreate()
	{
		mWikiApp = this;
		super.onCreate();

		Logger.setShowErrorEnabled(isDebugEnabled());
//		Logger.setEnabled(true);

		if(isDebugEnabled())
		{
			PreferencesManager.getInstance().setLoggingEnabled(true);
		}

		//We have a developer option that enables logging in production.
		//Therefore, we separate the enabling and disabling of it into this
		//if statement.
		if(PreferencesManager.getInstance().isLoggingEnabled())
		{
			Logger.setEnabled(isDebugEnabled());
			Logger.v("BRIEF", "yes this is logging");

		}

		//TODO fix debug flag
		Toaster.setEnabled(isDebugEnabled());
		Toaster.setEnabled(true);

		// Set a default expiration for hardcoded data if none is set (for testing)
		PreferencesManager prefsManager = PreferencesManager.getInstance();
		if (prefsManager.getHardcodedDataExpiration() == 0L) {
			Calendar calendar = Calendar.getInstance();
			calendar.setTime(new Date());
			calendar.add(Calendar.DAY_OF_YEAR, 7); // Expires in 7 days
			long defaultExpirationTimestamp = calendar.getTimeInMillis();
			prefsManager.setHardcodedDataExpiration(defaultExpirationTimestamp);
			Logger.i(TAG, "No hardcoded data expiration set. Defaulting to 7 days from now: " + new Date(defaultExpirationTimestamp));
		} else {
			Logger.i(TAG, "Hardcoded data expiration is already set to: " + new Date(prefsManager.getHardcodedDataExpiration()));
		}
	}

	public static boolean isDebugEnabled()
	{
		return BuildConfig.DEBUG;
	}

}

File: app/src/main/java/com/techventus/wikipedianews/WikiApplication.kt
package com.techventus.wikipedianews

import android.app.Application
import com.techventus.wikipedianews.logging.Logger
import com.techventus.wikipedianews.logging.Toaster
import com.techventus.wikipedianews.manager.PreferencesManager
import dagger.hilt.android.HiltAndroidApp
import timber.log.Timber

/**
 * Brief Application class with Hilt dependency injection.
 *
 * Modernization notes:
 * - Added @HiltAndroidApp for dependency injection
 * - Removed static singleton pattern (use Hilt injection instead)
 * - Integrated Timber for logging
 * - Fixed debug flag logic (removed TODO)
 */
@HiltAndroidApp
class WikiApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        // Initialize Timber logging
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
            Timber.d("Brief app starting in DEBUG mode")
        }

        // Legacy Logger setup for gradual migration
        Logger.setShowErrorEnabled(BuildConfig.DEBUG)

        if (BuildConfig.DEBUG) {
            PreferencesManager.getInstance().isLoggingEnabled = true
        }

        // Developer option for logging in production
        if (PreferencesManager.getInstance().isLoggingEnabled) {
            Logger.setEnabled(BuildConfig.DEBUG)
            Logger.v("BRIEF", "Legacy logging enabled")
            Timber.v("Legacy logging enabled")
        }

        // Fixed: Enable Toaster only in debug builds (removed duplicate line)
        Toaster.setEnabled(BuildConfig.DEBUG)
    }

    companion object {
        private val TAG = WikiApplication::class.java.simpleName

        val isDebugEnabled: Boolean
            get() = BuildConfig.DEBUG
    }
}
